apps/api-gateway/src/module.api/address-book.controller.ts

```diff
@@ -116,4 +117,25 @@ export class AddressBookController {
         )
     )
   }
+
+  @UseGuards(JwtAuthGuard)
+  @Get('address/:pin')
+  async getAddressByPin (
+    @Param('pin') pin: number,
+      @CurrentUser() user: User
+  ): Promise<PinAddressI> {
```

Comment:
Do not pass the userId as a payload. Remember the person using this endpoint is not the user with this address right? The PIN will do